frontend-react/src/utils/Analytics.ts

```diff
@@ -0,0 +1,19 @@
+import { IEventTelemetry } from "@microsoft/applicationinsights-web";
+
+import { getAppInsights } from "../TelemetryService";
+
+export enum EventName {
+    TABLE_FILTER = "Table Filter",
+}
+
+export const trackAppInsightEvent = (
+    eventName: string = "",
+    eventData: any
+) => {
+    const appInsights = getAppInsights();
+    const telemetryEvent: IEventTelemetry = {
+        name: eventName,
+    };
+
+    if (eventName !== "") appInsights?.trackEvent(telemetryEvent, eventData);
```

Comment:
Out of curiosity, is there any difference on the AI side in specifying `eventData` as `customProperties` versus `properties` in the IEventTelemetry object?

```jsx
appInsights?.trackEvent({
    name: telemetryEvent
    properties: eventData
});
```

Reply:
No difference. I'll change mine to be consistent with TelemetryService